### PR TITLE
TrayPublisher: added persisting of last selected project

### PR DESCRIPTION
**Pull request by kalisp**
## Brief description
Last selected project is preselected in Tray UI to make selection quicker.

## Additional info
Last selected project is stored in .json in app folder - eg. next to unzipped version zip.

## Testing notes:
1. open Tray Publisher and select project
2. close Tray Publisher - you don't need to publish anything
3. open TP again - last project should be preselected

### CODE REVIEW

File: openpype/tools/traypublisher/window.py [hunk 1]
```diff
@@ -90,6 +109,16 @@ def __init__(self, publisher_window):
...
     def showEvent(self, event):
         self._projects_model.refresh()
+
+        setting_registry = TrayPublisherRegistry()
+        project_name = setting_registry.get_item("project_name")
+        if project_name:
+            index = self._projects_model.get_index(project_name)
+            if index:
+                mode = QtCore.QItemSelectionModel.Select | \
+                       QtCore.QItemSelectionModel.Rows
```
Comment (iLLiCiTiT) [inline openpype/tools/traypublisher/window.py:119]:
```suggestion
                mode = (
                    QtCore.QItemSelectionModel.Select
                    | QtCore.QItemSelectionModel.Rows)
```

File: openpype/tools/utils/models.py [hunk 2]
```diff
@@ -330,6 +330,19 @@ def refresh(self):
         if new_items:
             root_item.appendRows(new_items)
 
+    def get_index(self, project_name):
```
Comment (iLLiCiTiT) [inline openpype/tools/utils/models.py:333]: I think find would be better in this case.
```suggestion
    def find_project(self, project_name):
```

File: openpype/tools/traypublisher/window.py [hunk 3]
```diff
@@ -90,6 +109,17 @@ def __init__(self, publisher_window):
 
     def showEvent(self, event):
         self._projects_model.refresh()
+
+        setting_registry = TrayPublisherRegistry()
+        project_name = setting_registry.get_item("project_name")
+        if project_name:
+            index = self._projects_model.get_index(project_name)
+            if index:
```
Comment (iLLiCiTiT) [inline openpype/tools/traypublisher/window.py:117]: There is missing conversion to proxy index. (Changes are based on [these changes](https://github.com/pypeclub/OpenPype/pull/3867/files#diff-1ecdf4cf8aaa62809627f8d57a300cb7a04414f370adaa27405e39bb6712d429R96) )
```suggestion
            index = None
            src_index = self._projects_model.get_index(project_name)
            if src_index is not None:
                index = self._projects_proxy.mapFromSource(src_index)
            if index:
```

**Review by mkolar — APPROVED** [review]
Works well from user perspective. Happy to merge as soon as code review is passed

File: openpype/tools/traypublisher/window.py (hunk 3 above)
Comment (kalisp) [reply openpype/tools/traypublisher/window.py]: Updated

**Review by iLLiCiTiT — APPROVED** [review]
Approved

**Pull request merged**